src/main/java/github/scarsz/discordsrv/api/SlashCommandManager.java

```diff
@@ -0,0 +1,133 @@
+/*-
+ * LICENSE
+ * DiscordSRV
+ * -------------
+ * Copyright (C) 2016 - 2022 Austin "Scarsz" Shapiro
+ * -------------
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-3.0.html>.
+ * END
+ */
+
+package github.scarsz.discordsrv.api;
+
+import github.scarsz.discordsrv.DiscordSRV;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import net.dv8tion.jda.api.exceptions.ErrorResponseException;
+import net.dv8tion.jda.api.exceptions.RateLimitedException;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import net.dv8tion.jda.api.requests.ErrorResponse;
+import net.dv8tion.jda.api.requests.restaction.CommandListUpdateAction;
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+public class SlashCommandManager {
+    @Getter
+    private boolean updated = false;
+    private final Map<Plugin, PluginCommands> commandsMap = new HashMap<>();
+
+    public void reloadSlashCommands() {
+        CommandListUpdateAction action = DiscordSRV.getPlugin().getMainGuild().updateCommands();
```

Comment:
I think it'd be a good idea to avoid using bulk update and instead adding, updating & deleting commands as necessary since the rate limit is per command that is added/updated/deleted so this would quickly drain the limit.

`getMainGuild` might return null. 

Also worth considering what use cases people have in mind, some times people add DiscordSRV to multiple servers, and might want to use slash commands in other guilds too

Reply:
But.. Rate limit is per request.. Not commands updated/edited/added, 100/guild/day it's per request

Reply:
![image](https://user-images.githubusercontent.com/20226336/177048658-69d0f610-0981-44a5-b3ee-6692bdf583f4.png)

[source](https://discord.com/developers/docs/interactions/application-commands#registering-a-command)
Not entirely sure about the scope of 'create' here, but non the less this is a daily limit

Reply:
This only includes unique updates because `updateCommands()` is idempotent for duplicate commands. Using `upsertCommand()` is a very bad idea, and should be avoided at all costs.

If you have 20 commands, you can use `updateCommands()` and set them all with only 1 request, instead of 20.

Reply:
Alright, so the way registering works has changed since I last used it, so that should actually be fine. The other comments still apply

Reply:
This hasn't changed. The only difference is with Application Permissions V2 where global commands now show up instantly instead of ~30 minutes.

Reply:
how do you want the big warning to look like when an error happens when registering the commands in a few servers? like server 1 missing scope, server 2 rate limited, server 3 unknown error... (in case you want registration in every guild)

Reply:
Is the rate limit for 10 commands in a single `updateCommands()` call the same as 10 individual `upsertCommand()` calls?

Reply:
`<guild name>: <error message>` 1 line per server or something like that